### DISCUSSION TITLE
Failed to pull singularity image?

### DISCUSSION THREAD

**Discussion by zenmck** [nextflow-io/nextflow#5369] [Q&A]
Howdy,

I am currently using Nextflow to help develop a pipeline for seed health testing, and this is the current issue we are running into:
```
Caused by:
  Failed to pull singularity image
    command: singularity pull  --name community.wave.seqera.io-library-krakentools-1.2--5332070a89999eef.img.pulling.1728061063849 oras://community.wave.seqera.io/library/krakentools:1.2--5332070a89999eef > /dev/null
    status : 255
    hint   : Try and increase singularity.pullTimeout in the config (current is "20m")
    message:
      FATAL:   While pulling image from oci registry: error fetching image to cache: failed to get checksum for oras://community.wave.seqera.io/library/krakentools:1.2--5332070a89999eef: ORAS SIF image should have a single layer, found 14



 -- Check '.nextflow.log' file for details

The pipeline encountered an error and did not finish successfully
WARN: Killing running tasks (1)
```

Yet, when I go to check the '.nextflow.log' file or the '.command.error' I do not see anywhere we would have put "20m" for the singularity.pullTimeOut? And even in the config file we have I do not see "20m"

Best,

Zen

**Comment by pditommaso** [comment]
Please provide the `.nextflow.log` file

**Comment by zenmck** [comment]
[nextflow(1).log](https://github.com/user-attachments/files/17261921/nextflow.1.log)
Does this work? Sorry, this is my first time using GitHub issues outside of a classroom setting.

**Reply by pditommaso** [reply]
Why is the container prefixed with `oras://`? doesn't look to be a singularity container image

**Reply by zenmck** [reply]
Correct, we are using Seqera containers for each of the processes we plan on using, as singularity is usually compatible with all Docker images, and my professor recommended this open-source platform.  When the container is ready it looks like this, specifically with fastp, "community.wave.seqera.io/library/fastp:0.23.4--4d9e23447c67e79e" and then I add the "oras://" as I believe that is what my advisor did as well?

**Reply by ewels** [reply]
That's not quite how it works. This is the Docker image:
```
community.wave.seqera.io/library/fastp:0.23.4--4d9e23447c67e79e
```

You can use this with Singularity directly and it'll convert the Docker image to a Singularity image. But you can't just add the `oras://` protocol to it like that.

If you want a Singularity image, you need to request it specifically in the interface:

<img src="https://github.com/user-attachments/assets/a459d087-057c-4747-b9d2-e4d3ba322b49" width=300>


Then a bespoke Singularity image will be built for you, and that comes with the oras protocol:

```
oras://community.wave.seqera.io/library/fastp:0.23.4--4ea6310369653ec7
```

<img src="https://github.com/user-attachments/assets/d9629524-ead6-4573-b592-702b47765aeb" width=500>

Note that the container name is not exactly the same though, the hash is different. This is because this is a totally different image, built from scratch as a native Singularity image.

Hope that makes sense!

**Reply by zenmck** [reply]
Howdy,

Yes, that does make a lot of sense, thank you for the clarification!
